Move proxy to prow config generation

howardjohn opened the pull request:
This PR moves the proxy to the new prow config generation. I made this
enabled only for the "prow-stage" branch so we can do send a dummy PR to
validate I didn't break anything.

Benefits:
* Proxy follows same conventions of other repos (in terms of naming,
testgrid, branching, etc)
* Config should be much simpler to understand
* Resolves some issues with duplicate names, confusing context, etc

The only effective changes after this is the change to the job names. I
am open to renaming, the reason I didn't keep the job names as before is
we would have ended up with names like
proxy-postsubmit_proxy_postsubmit, due to the naming convention of
test_repo_jobtype.

bianpengyuan commented on prow/config/jobs/proxy.yaml:
```diff
@@ -0,0 +1,43 @@
+jobs:
+- name: test
+  type: presubmit
+  repos: [istio/istio@master]
```
Should these be istio/proxy?

howardjohn replied:
`repos` are extra repos to checkout, proxy is implicitly checked out at the PR commit for presubmit and the commited PR for for postsubmit.

The proxy jobs all explicitly configured this before, so I assumed they depend on something in istio/istio, is that not needed?

bianpengyuan replied:
I don't think it is necessary. Maybe @PiotrSikora would have better idea. I think it is fine to remove it.

howardjohn replied:
Well this is just for a dummy test branch anyways, so I'll just remove it then add it back later if we need it. No risk of breaking something

bianpengyuan replied:
sg.

howardjohn committed:
Move proxy to prow config gen

This PR moves the proxy to the new prow config generation. I made this
enabled only for the "prow-stage" branch so we can do send a dummy PR to
validate I didn't break anything.

Benefits:

Proxy follows same conventions of other repos (in terms of naming,
testgrid, branching, etc)
Config should be much simpler to understand
Resolves some issues with duplicate names, confusing context, etc
The only effective changes after this is the change to the job names. I
am open to renaming, the reason I didn't keep the job names as before is
we would have ended up with names like
proxy-postsubmit_proxy_postsubmit, due to the naming convention of
test_repo_jobtype.

howardjohn committed:
Fix
